### PR TITLE
Fixes issue with rocket soldiers wandering

### LINKED ISSUE #12368: Suicide behavior of rocket soldiers
Rocket soldiers tend to get themselves killed by wandering away from the group to chase down enemy ground units. This is very annoying.

This occurs because the AutoTarget ScanRadius defaults to the unit's max weapon range. Since the rocket soldier has two different weapons with two different ranges, it uses the one with the longest range. This makes rocket soldiers chase ground units with a scan radius of 7.5 cells (AA range) instead of 5 cells (AG range). 
Setting their ScanRadius to 5 cells should prevent this excessive chase behavior.

OpenRA\mods\ra\rules\infantry.yaml
E3:
+AutoTarget:
++**ScanRadius: 5**

Source infantry chase behavior:
OpenRA\mods\ra\rules\default.yaml
^Soldier:
+AutoTarget:
++ScanRadius: -1 ([default to maximum weapon range](https://github.com/OpenRA/OpenRA/wiki/Traits#autotarget))

### PR DESCRIPTION
**Pull request by RobAWilkinson**
Fixes #12368

### CODE REVIEW

**Comment by FrameLimiter** [comment]
Hi RobAWilkinson,

Check line 268 and 280 in your edited defaults.yaml -- you have duplicate AutoTarget: traits.
Since the AutoTarget: is already on line 268, and the default ScanRadius value is automatically "-1". Nothing should need to be edited in the default.yaml.
The E3 edit looks good :+1:  
Thanks for making the pull request. :)

File: mods/ra/rules/defaults.yaml [hunk 1]
```diff
@@ -277,6 +277,8 @@
 		StandSequences: stand,stand2
 	DetectCloaked:
 		CloakTypes: Hijacker
+	AutoTarget:
```
Comment (GraionDilach) [inline mods/ra/rules/defaults.yaml]: You already have an `AutoTarget` in line 268.

File: mods/ra/rules/defaults.yaml [hunk 2]
```diff
@@ -277,6 +277,8 @@
...
+		ScanRadius: -1
```
Comment (GraionDilach) [inline mods/ra/rules/defaults.yaml]: You don't even need this, since this is the default value. The default.yaml changes should be reverted.

Comment (RobAWilkinson) [inline mods/ra/rules/defaults.yaml]: awesome, my mistake, reverted and pushed so its just the `infantry.yaml` now

**Comment by Phrohdoh** [comment]
@RobAWilkinson These commits will need to be squashed.

Ask in IRC if you need help doing so.

**Comment by RobAWilkinson** [comment]
@Phrohdoh squashed :+1: for keeping history clean.

**Review by chrisforbes — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5cd63f464dba99316b0d5094dd8e21f009f19e0d...ad60e659907c548723302a211877c5e30a660595)

**Comment by pchote** [comment]
This was merged onto master instead of bleed!

**Comment by pchote** [comment]
I have cherry-picked the commit (now e337048a97d9c3f5324657872a144aeb10c7bae2) to bleed and reverted the master branch.

**Comment by RobAWilkinson** [comment]
@pchote oh wow! thanks

**Comment by reaperrr** [comment]
> This was merged onto master instead of bleed!

Eeek, sorry. I saw that his branch was called master, but not that it was filed *against* master.

**Comment by Phrohdoh** [comment]
Does the new branch protection stuff guard against merging into master?
(afaik we only push directly to master when creating a release)

**Pull request merged**